src/main/java6/net/finmath/marketdata/model/curves/ForwardCurveFromDiscountCurve.java

```diff
@@ -45,92 +46,50 @@
 	 * The date t + p is generated from the paymentOffsetCode using the provided paymentOffsetBusinessdayCalendar and paymentOffsetDateRollConvention.
 	 * 
 	 * @param name The name under which the forward curve can be referenced.
-	 * @param discountCurveName The discount curve used for calculation of the forward.
+	 * @param baseDiscountCurveName The discount curve used for calculation of the forward.
 	 * @param referenceDate The reference date used in the interpretation of times (i.e., the referenceDate where t=0).
 	 * @param paymentOffsetCode The payment offset. If null, the parameter p has to be provided to the getForward method.
 	 * @param paymentOffsetBusinessdayCalendar The calendar used to generate the payment date from the paymentOffetCode.
 	 * @param paymentOffsetDateRollConvention The date roll convention used to generate the payment date from the paymentOffsetCode.
 	 * @param daycountScaling The scaling factor applied to the paymentOffset measured in ACT/365.
 	 * @param periodOffset An offset in ACT/365 applied to the fixing to construct the period start (the negative of the fixingOffset of the period).
 	 */
-	public ForwardCurveFromDiscountCurve(String name, String discountCurveName, LocalDate referenceDate, String paymentOffsetCode, BusinessdayCalendarInterface paymentOffsetBusinessdayCalendar, BusinessdayCalendarInterface.DateRollConvention paymentOffsetDateRollConvention, double daycountScaling, double periodOffset) {
-		super(name, referenceDate, paymentOffsetCode, paymentOffsetBusinessdayCalendar, paymentOffsetDateRollConvention, discountCurveName);
+	public ForwardCurveFromDiscountCurve(String name, String baseDiscountCurveName, LocalDate referenceDate, String paymentOffsetCode, BusinessdayCalendarInterface paymentOffsetBusinessdayCalendar, BusinessdayCalendarInterface.DateRollConvention paymentOffsetDateRollConvention, double daycountScaling, double periodOffset) {
+		super(name+nameSuffix, referenceDate, paymentOffsetCode, paymentOffsetBusinessdayCalendar, paymentOffsetDateRollConvention);

```

Comment:
I belive this change might generate major issues. The JavaDoc says:

> name is the name under which the forward curve can be reference

but the class is internally modifying the name - ALWAYS. The user will not be able to get the curve unter its name. He had to check for getName(). This is an issue. Consider the case where you like to create a curve under the name "EUR 6M", using a reference to a hidden discount curve.

I do not that calibrate curves uses the "trick" with the suffix, but why force to have all ForwardCurveFromDiscountCurve behave like that.?